### ISSUE TITLE
QR code ECC failure

### ISSUE THREAD

**Issue by claudiofelber** [dlbeer/quirc#88]
I have an example of a QR code which Quirc is not able do decode. It can locate it within the image but fails later with an ECC error when decoding the data. I checked the code with the Quirc inspect tool and the pixels appear to have been read correctly. I wonder if the problem is related to the small overlayed image in the center (it's required by the Swiss QR bill standard). Other QR code readers such as ZXing or the built-in QR code reader on iOS are able to decode the image correctly. boofcv on the other hand also fails to decode the QR code data.
![qrbill_failed](https://user-images.githubusercontent.com/339950/93205496-3107d100-f758-11ea-8617-b109e7771c03.png)

**Comment by claudiofelber** [comment]
In the meantime I have figured it out by myself. The QR code is flipped horizontally. According to the ISO 18004:2015 QR code specification this is an optional feature and is probably used to recognize QR codes posted on transparent surfaces, such as a glass door, where it should be possible to scan the QR code from both sides. Quirc does not handle this case. I think Quirc is based on the ISO 18004:2000 QR code specification which does not yet mention image mirroring.

I have implemented a Quirc extension handling this use case. As soon as my currently pending PR https://github.com/dlbeer/quirc/pull/87 has been accepted and integrated, I will add another one with the aforementioned extension.
